app/controllers/benchmark_runs_controller.rb

```diff
@@ -30,6 +30,10 @@ def create
     benchmark_run.update_attributes(benchmark_run_params)
     benchmark_run.result = params[:benchmark_run][:result]
     benchmark_run.save!
+ 
+    if params[:commit_hash]
+      BenchmarkRegressionJob.perform_later(initiator, benchmark_type, benchmark_result_type, category)
```

Comment:
I believe you can pass in the benchmark_run created here and it'll make things alot easier